developer/src/kmc/src/keyman/compiler/metadata-compiler.ts

```diff
@@ -0,0 +1,55 @@
+import KMXFile, { KEYBOARD } from "../kmx/kmx";
+import { KMXPlusData } from "../kmx/kmx-plus";
+import CompilerOptions from "./compiler-options";
+
+const KEYMAN_VERSION = require("@keymanapp/keyman-version").KEYMAN_VERSION;
+
+export default class KMXPlusMetadataCompiler {
+  /**
+   * Look for metadata fields in the KMXPlus data and copy them
+   * through to the relevant KMX stores
+   * @param kmxplus const KMXPlusData
+   */
+  public static addKmxMetadata(kmxplus: KMXPlusData, keyboard: KEYBOARD, options: CompilerOptions): void {
+    // Order of stores is not significant by kmx spec, but kmxplus compiler will
+    // always store according to dwSystemID binary order for non-zero
+    // dwSystemID, then by dpName binary order, and finally by dpString binary
+    // order
+
+    // TSS_NAME = 7
+    // TSS_COMPILEDVERSION = 20
+    // TSS_KEYBOARDVERSION = 36
+    // TSS_TARGETS = 38
+
+    // TSS_NAME: User friendly name of keyboard
+    keyboard.stores.push({
+      dpName: '&NAME',
+      dpString: kmxplus.name?.names?.[0]?.value ?? 'unknown',  // Empty name should not happen, so ok to use 'unknown' here
```

Comment:
could be unnamed

Reply:
It probably can't happen unless we abuse metadata-compiler in our code, so I'll leave this one :grin:.